cupyx/scipy/special/_stats_distributions.py

```diff
@@ -32,6 +32,53 @@
     ''')
 
 
+log_ndtr_definition = """
+
+#define NPY_SQRT1_2   0.707106781186547524400844362104849039  /* 1/sqrt(2) */
+
+template <typename T>
+static __device__ T log_ndtr(T x)
+{
+    T t = x * NPY_SQRT1_2;
+    if (x < -1.0) {
+        return log(erfcx(-t) / 2) - t * t;
+    } else {
+        return log1p(-erfc(t) / 2);
+    }
+}
```

Comment:
We can change this algorithm to `log(ndtr)`. But I see scipy has used this algorithm, due to numerical precision issues. Ref: https://github.com/scipy/scipy/pull/15187. Is this okay? Thanks